### PR TITLE
Update release workflow and prepare for next release

### PR DESCRIPTION
**Pull request by gotmax23**

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -180,18 +180,19 @@ def bump(session: nox.Session):
                 f"Either {fragment_file} must already exist, "
                 "or two positional arguments must be provided."
             )
-    install(session, "antsibull-changelog[toml]", "hatch")
+    # Needs newer antsibull-changelog for hatch version auto-detection support
+    install(session, "antsibull-changelog >= 0.24.0", "hatch")
```
Comment (felixfontein) [inline noxfile.py:184]:
```suggestion
    install(session, "antsibull-changelog[toml] >= 0.24.0", "hatch")
```

**Comment by felixfontein** [comment]
This needs to be backported manually, since the MD changelog will look different there.

File: noxfile.py (hunk 1 above)
Comment (gotmax23) [reply noxfile.py:184]: Ah, right...

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
> This needs to be backported manually, since the MD changelog will look different there.

Actually it does not, since we never did a 3.x.y release so far.

But I also noticed I forgot about the `rebase_merge` label. Sorry :-(

**Pull request merged**